src/upgrade.h

```diff
@@ -0,0 +1,37 @@
+#ifndef ZCASH_UPGRADE_H
+#define ZCASH_UPGRADE_H
+
+#include <univalue.h>
+
+#include <string>
+
+struct CUpgrade
+{
+public:
+    std::string upgradeId;
+    std::string status;
+
+    // Block height of network upgrade
+    int64_t ruleChangeHeight;
+    int64_t txEmbargoStartHeight;
+    int64_t txEmbargoEndHeight;
+
+    // Additional information
+    std::string info;
+
+    CUpgrade(std::string upgradeIdIn, std::string statusIn, int64_t ruleChangeHeightIn, int64_t txEmbargoStartHeightIn, int64_t txEmbargoEndHeightIn, std::string infoIn)
+    {
+        upgradeId = upgradeIdIn;
+        status = statusIn;
+        ruleChangeHeight = ruleChangeHeightIn;
+        txEmbargoStartHeight = txEmbargoStartHeightIn;
+        txEmbargoEndHeight = txEmbargoEndHeightIn;
+        info = infoIn;
```

Comment:
The idiomatic way to write this in C++ is:

```
CUpgrade(std::string upgradeIdIn, std::string statusIn, int64_t ruleChangeHeightIn,
         int64_t txEmbargoStartHeightIn, int64_t txEmbargoEndHeightIn, std::string infoIn) :
    upgradeId(upgradeIdIn),
    status(statusIn),
    ruleChangeHeight(ruleChangeHeightIn),
    txEmbargoStartHeight(txEmbargoStartHeightIn),
    txEmbargoEndHeight(txEmbargoEndHeightIn),
    info(infoIn)
{
}
```

Reply:
This doesn't work, and everywhere else I've seen attributes set by the `=` sign. Is there a link to a reference or example I could refer to?

Reply:
Ok I changed it